docs/protocol/custom_extension.md

```diff
@@ -0,0 +1,76 @@
+---
+sidebar_position: 5
+---
+
+# Custom Extensions
+
+Notary attestations can be extended with custom fields defined by the `Prover`. This allows applications to attach additional data to attestations in a verifiable way. For now, the `Notary` simply includes these fields in the signed attestation without validation. In the future, a plugin system will allow developers to add custom logic to the `Notary` for verifying this data. Until then, such validation requires modifying the Notary code directly.
```

Comment:
This is the introduction. So it is OK to give a preview of what is coming later in the document.

```suggestion
Notary attestations can be extended with custom fields defined by the `Prover` or the `Notary`. This allows applications to attach additional data to attestations in a verifiable way.

For now, the default notary server implementation simply includes these fields in the signed attestation without validation. In the future, a plugin system will allow developers to add custom logic to the notary server for verifying this data. Until then, such validation requires a custom implementation of the Notary API.
```

Reply:
1st paragraph looks good. 2nd one can be simplified I think, i.e.

`
For now, the default notary server implementation only supports including custom data requested by the Prover in the attestation without any validation. In the future, a plugin system will allow developers to add custom logic to the notary server for adding or validating these data.
`

Reply:
👍